core/logger/seelog.go

```diff
@@ -17,25 +17,93 @@ limitations under the License.
 package logger
 
 import (
+	"context"
 	"fmt"
 	"strings"
 	"sync"
 
+	"github.com/cihub/seelog"
 	log "github.com/cihub/seelog"
 	"github.com/dan-drl/framework/core/config"
 	"github.com/dan-drl/framework/core/env"
 	"github.com/dan-drl/framework/core/util"
 	"github.com/ryanuber/go-glob"
+	"go.elastic.co/apm"
 )
 
 var file string
 var loggingConfig *config.LoggingConfig
 var l sync.Mutex
 var e *env.Env
 
+func init() {
+	err := log.RegisterCustomFormatter("APM", createAPMFormatter)
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+}
+
+func createAPMFormatter(params string) log.FormatterFunc {
+	return func(message string, level log.LogLevel, logContext log.LogContextInterface) interface{} {
+		if logContext.CustomContext() != nil {
+			context := logContext.CustomContext().(context.Context)
+			if context != nil {
+				return fmt.Sprintf("[%+v]", apm.TraceFormatter(context))
+			}
+		}
+
+		return ""
+	}
+}
+
+// NewLogger create a new seelog logger based on current config
+func NewLogger() seelog.LoggerInterface {
+	consoleWriter, _ := NewConsoleWriter()
+
+	format := "[%Date(01-02) %Time] [%LEV] [%File:%Line] %APM %Msg%n"
+	formatter, err := log.NewFormatter(format)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	rollingFileWriter, _ := NewRollingFileWriterSize(file, rollingArchiveNone, "", 250000000, 5, rollingNameModePostfix)
+	bufferedWriter, _ := NewBufferedWriter(rollingFileWriter, 10000, 1000)
+
+	l, _ := log.LogLevelFromString(strings.ToLower(loggingConfig.LogLevel))
+	pushl, _ := log.LogLevelFromString(strings.ToLower(loggingConfig.PushLogLevel))
+
+	//logging receivers
+	receivers := []interface{}{consoleWriter, bufferedWriter}
```

Comment:
Is there a way to disable writing to disk? The way the daemonsets in k8s are being setup, writing to disk will not be necessary in the near future.

Reply:
Ah good point! That'll make the new loggers easier to generate. I'm assuming I can just use the console writer then. I'll try that out.